src/rgw/rgw_sal_motr.cc

```diff
@@ -1288,9 +1288,21 @@ int MotrBucket::list(const DoutPrefixProvider *dpp, ListParams& params, int max,
         auto iter = vals[i].cbegin();
         ent.decode(iter);
         if (params.list_versions || ent.is_visible()) {
+          if (ent.key.name == params.marker.name &&
```

Comment:
Could you add some comments to explain what these conditions do (including other in your patch)? those conditions are not easy to follow for some one who first read the code.

The patch looks fine to me, just this minor thing.